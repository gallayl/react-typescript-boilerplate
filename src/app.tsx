import React from 'react'
import { Button, CssBaseline, Typography } from '@material-ui/core'

export const App: React.FunctionComponent = () => {
  return (
    <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'center', width: '100%', height: '100%' }}>
      <CssBaseline />
      <Typography variant="h3">
        <Button>anyÃ¡d ğŸ’–</Button>
      </Typography>
    </div>
  )
}
