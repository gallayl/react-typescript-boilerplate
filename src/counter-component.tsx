import React, { useEffect, useState } from 'react'

export const Counter: React.FunctionComponent<{ default: number }> = props => {
  const [currentCount, setCurrentCount] = useState(props.default)

  useEffect(() => {
    setCurrentCount(props.default)
  }, [props.default])

  return (
    <span>
      <button onClick={() => setCurrentCount(currentCount + 1)}>+</button>
      <input
        defaultValue={`${currentCount.toString()}/${props.default.toString()}`}
        style={{
          color: currentCount === props.default ? 'darkgreen' : 'red',
        }}
      />
      <button onClick={() => setCurrentCount(currentCount - 1)}>-</button>
    </span>
  )
}
