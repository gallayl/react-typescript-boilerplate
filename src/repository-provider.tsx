import React, { createContext, useEffect, useState } from 'react'
import { FormsAuthenticationService, LoginState, Repository } from '@sensenet/client-core'
import { RepositoryConfiguration } from '@sensenet/client-core/dist/Repository/RepositoryConfiguration'
import { CircularProgress } from '@material-ui/core'
import { LoginForm } from './components/login-form'

export const RepositoryContext = createContext(new Repository())

export const lastRepositoryKey = 'sn-boilerplate-last-repository'

export const RepositoryProvider: React.FunctionComponent<
  Omit<Partial<RepositoryConfiguration>, 'repositoryUrl'>
> = props => {
  const [currentRepoUrl, setCurrentRepoUrl] = useState(localStorage.getItem(lastRepositoryKey) || '')
  const [currentRepo, setCurrentRepo] = useState(new Repository({ ...props, repositoryUrl: currentRepoUrl }))

  const [loginState, setLoginState] = useState(LoginState.Unknown)

  useEffect(() => {
    localStorage.setItem(lastRepositoryKey, currentRepoUrl)
    const repo = new Repository({ ...props, repositoryUrl: currentRepoUrl })
    FormsAuthenticationService.Setup(repo)
    setCurrentRepo(repo)
    const observable = repo.authentication.state.subscribe(state => setLoginState(state), true)
    return () => observable.dispose()
  }, [currentRepoUrl, props])

  return (
    <RepositoryContext.Provider value={currentRepo}>
      {loginState === LoginState.Pending ? <CircularProgress /> : null}
      {loginState === LoginState.Authenticated ? props.children : null}
      {loginState === LoginState.Unauthenticated || loginState === LoginState.Unknown ? (
        <LoginForm
          onLogin={(username, password, repoUrl) => {
            setCurrentRepoUrl(repoUrl)
            currentRepo.authentication.login(username, password)
          }}
        />
      ) : null}
    </RepositoryContext.Provider>
  )
}
