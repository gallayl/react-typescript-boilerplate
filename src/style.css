/**
 * Global Style overrides
**/

ul[role='listbox'] {
  margin: 0;
  padding: 0;
}

a[href] {
  color: inherit;
}
