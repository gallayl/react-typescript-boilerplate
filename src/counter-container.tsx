import React, { useState } from 'react'
import { Counter } from './counter-component'

export const CounterContainer: React.FunctionComponent = () => {
  const [values, setValues] = useState<number[]>([])

  return (
    <div>
      <button
        onClick={() => {
          const newValues = []
          for (let index = 0; index < 1000; index++) {
            newValues.push(Math.round(Math.random() * 1000))
          }
          setValues([...newValues])
        }}>
        randomize
      </button>
      <div>
        {values.map((v, i) => (
          <Counter default={v} key={i} />
        ))}
      </div>
    </div>
  )
}
