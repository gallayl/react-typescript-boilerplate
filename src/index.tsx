import React from 'react'
import ReactDOM from 'react-dom'
import './style.css'
import { App } from './app'
import { RepositoryProvider } from './context/repository-provider'

/**
 * Initialize React
 */
ReactDOM.render(
  /** The RepositoryProvider will display a login form for non-authenticated users */
  <RepositoryProvider
    /** You can insert your additional repository settings here */
    sessionLifetime="expiration">
    <App />
  </RepositoryProvider>,
  document.getElementById('root'),
)
