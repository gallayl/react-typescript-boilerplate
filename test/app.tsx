import { shallow } from 'enzyme'
import React from 'react'
import { App } from '../src/app'

describe('Layout', () => {
  it('Matches snapshot', () => {
    const l = shallow(<App />)
    expect(l).toMatchSnapshot()
  })
})
