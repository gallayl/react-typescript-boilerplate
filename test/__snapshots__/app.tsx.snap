// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Matches snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <CssBaseline />
  <WithStyles(ForwardRef(Typography))
    variant="h3"
  >
    <WithStyles(ForwardRef(Button))>
      anyÃ¡d ðŸ’–
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Typography))>
</div>
`;
